### ISSUE TITLE
Always get e-mail / id code for a user account

### ISSUE THREAD

**Issue by loorm** [citizenos/citizenos-fe-old#588]
**Problem**
User register's with eID. Later, someone wants to invite the user into a topic. Sends the invite by e-mail. Because the original eID account did not have e-mail information, a new account is now created for the same user. This is a suboptimal situation.

**What is the user story?**
When I register as a user, I want to be able to enjoy the full functionality of Citizen OS.  Because I do not know, that e-mail is the way can be invited to topics, I want the system to ask for my e-mail, if I have not entered one (i.e. have created account with an eID).  I also want the system to make this mandatory and to verify my e-mail.

**Which feature needs to be updated and how?**
Account creation needs to be updated, so that:
- if a user registers account with eID, we ask for e-mail as well (mandatory field)
- if a user registers account with e-mail or social ID, offer up option to connect to eID as well (optional field)

**Additional information.**

- [x] user story

- [x] legal review

- [ ] UX update the views and fields

- [ ] API update and implementation

Preferably, it will not be a separate step and window, but integrated into the current flow.

**Comment by loorm** [comment]
Issue is now fully described and ready for @kevincrepin

**Comment by loorm** [comment]
Triage 30. Confirmed. Actually, it needs a user story and a legal review before UX. @kevincrepin, please hold until the first two steps are completed.

**Comment by KatiVellak** [comment]
@loorm please ping me once your user story is complete, the current user story didn´t raise any red flags so far. As the email field is meant to be mandatory, it would be good idea to have an information pop-up there to explain the reasons. And also to connect the eID as an option should have the explanation so that the user right away understand the need and importance.

**Comment by tiblu** [comment]
@loorm  Moved to "In preparation", does not look ready for development.

**Comment by loorm** [comment]
@KatiVellak User story is now complete (I edited the first comment, see above).

**Comment by loorm** [comment]
@kevincrepin I think we can proceed to views. If an explanatory text is needed, I propose the following copy: "You can only be invited to join discussions and votes by e-mail. Please enter your valid e-mail to proceed."

"If you have an eID (digital signature, mobile ID etc), you can connect it to your account now, in order to vote for decisions.

**Comment by loorm** [comment]
> @loorm please ping me once your user story is complete, the current user story didn´t raise any red flags so far. As the email field is meant to be mandatory, it would be good idea to have an information pop-up there to explain the reasons. And also to connect the eID as an option should have the explanation so that the user right away understand the need and importance.

@KatiVellak User story is now complete (I edited the first comment, see above).

**Comment by kevincrepin** [comment]
I added flows here: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=737%3A26045

When adding e-mail we explain why we ask this and also that the user should still verify their e-mail after they receive our e-mail :)

Syncing is not obligatory so can be skipped entirely

**Comment by loorm** [comment]
@kevincrepin Can we change "**sync** your account with eID" to "**connect** your account with eID"?

**Comment by loorm** [comment]
@anettlinno, I'm handing this issue over to you. We now have designs from Kevin. Still need a legal review from Kati before proceeding, I don't want to move to Triage before her opinion. You have the ball.

**Comment by kevincrepin** [comment]
Changed the copy in the meantime :)

**Comment by anettlinno** [comment]
I think that there are two sides in this issue: (1) connecting e-ID and email accounts and (2) updating "My account" information fields. 

Kevin has nicely solved with the pop-up how to ask email address after User has signed in with e-ID. There could be also explanatory text that User can manage and change her email under "My account". AND I would link here an issue #521 which is  about asking more information from Users (just FYI). 

My questions to @kevincrepin @tiblu @ilmartyrk  are: 
Can User connect her e-ID with multiple emails? IF yes then how the email confirmation flow works with the second/third/fourth email? The same? One email should also be "master" email that will be shown in voting results (if social-ID voting) and User should have an opportunity to choose "master" email. 
Should we also explain under "My account", which information will be provided for the Topic admins (ID number, email, name if we ask more then what else)?

AND Users who already have COS accounts should also have an opportunity to connect and manage their ID numbers and email addresses (connecting different accounts) under "My account".

**Comment by anettlinno** [comment]
Friendly reminder to @kevincrepin and @KatiVellak that we are looking for your input :) 

@kevincrepin we would need your beautiful design for the "My account" view where users can manage their connected accounts (ID number, email addresses and if you please check the #521 then also phone number). 

@KatiVellak we would need your legal advise here regarding connecting accounts. 

Thanks a lot in advance!

**Comment by KatiVellak** [comment]
@anettlinno I do not see any legal issues herein. When registering with the eID, we ask for the email and explain the reasons for it. The email in this case is mandatory because otherwise the user will have little to gain from our platform as email is needed to join the topic. 
When user joins by email, we offer as optional possibility to sync user's eID with his/her account, user's own decision. 
And taking into consideration the fact that individuals nowadays have several emails for several different purposes, then I do encourage us to right away think of the solution to enable the user to connect his/her account with several emails, subject to @anettlinno comment on Sep 23, 2021.

**Comment by anettlinno** [comment]
Triage 51. Discussed and decided that we will ask email addresses from existing users next time when they log in with their eID. With new users who login with eID we ask for e-mail as well (mandatory field) and if a user registers account with e-mail or social ID, we offer option to connect to his/her eID (optional field).

Sending to development to implement together with #239 and #670. Est. dev. time 3 hours.

**Comment by ilmartyrk** [comment]
Going to release this update in 2 waves. 
- [x] 1) Add require e-mail flow
- [x] 2) Add options to connect eID

**Comment by ilmartyrk** [comment]
So I've added the API part and was adding the FE view for connecting with eID after registering. But I ran into many different questions.
1) We should have a view under My account to add different connections. There is something similar already in the history https://projects.invisionapp.com/d/main?origin=v7#/console/9829159/253611417/preview?scrollOffset=10400 should we go with that or get an update from  @kevincrepin ?
2) Also as there is possibility that user has multiple accounts connected with personal Id should we make a possibility to switch between accounts. 
 2.1) If we do, I believe it should be strictly for only users who have authenticated using Smart-ID, Mobiil-ID or ID-card
3) Should we separate profile e-mail and login email. Meaning that under profile the e-mail is what we use to contact the person (send invites, etc.) and the login e-mail will be the e-mail used for logging in

Anyway in the scope of this issue, I believe that we should also update my account view and add possibilty to add connections from there, but I would not allow deleting the connections with personal ID, only allow removing google and facebook connections.

What do you thin  @anettlinno  @kevincrepin

**Comment by ilmartyrk** [comment]
I've added the views. Currently the connect eID view is displayed when terms and conditions view is displayed and only if no connection exists. This is because all new users will see this view. I am still thinking of adding possible update to my account view. Have created a new issue for that https://github.com/citizenos/citizenos-fe/issues/931
